pkg/sql/pg_catalog.go

```diff
@@ -2762,19 +2762,65 @@ CREATE TABLE pg_catalog.pg_type (
 				}
 
 				// Now generate rows for user defined types in this database.
-
 				return forEachTypeDesc(ctx, p, dbContext, func(_ *sqlbase.ImmutableDatabaseDescriptor, _ string, typDesc *sqlbase.ImmutableTypeDescriptor) error {
-					typ, err := typDesc.MakeTypesT(
-						tree.NewUnqualifiedTypeName(tree.Name(typDesc.GetName())),
-						p.makeTypeLookupFn(ctx),
-					)
+					typ, err := createTypeFromDesc(ctx, p, typDesc)
 					if err != nil {
 						return err
 					}
 					return addPGTypeRow(h, nspOid, typ, addRow)
 				})
 			})
 	},
+	indexes: []virtualIndex{
+		{
+			partial: false,
+			populate: func(ctx context.Context, constraint tree.Datum, p *planner, db *sqlbase.ImmutableDatabaseDescriptor,
+				addRow func(...tree.Datum) error) (bool, error) {
+
+				h := makeOidHasher()
+				nspOid := h.NamespaceOid(db, pgCatalogName)
+				coid := tree.MustBeDOid(constraint)
+				ooid := oid.Oid(int(coid.DInt))
+
+				// Check if it is a predefined type.
+				typ, ok := types.OidToType[ooid]
+				if ok {
+					if err := addPGTypeRow(h, nspOid, typ, addRow); err != nil {
+						return false, err
+					}
+					return true, nil
+				}
+
+				// Check if it is a user defined type.
+				id := sqlbase.ID(types.UserDefinedTypeOIDToID(ooid))
+				// TODO (rohany): This access should go through the desc.Collection.
+				typDesc, err := sqlbase.GetTypeDescFromID(ctx, p.txn, keys.SystemSQLCodec, id)
+				if err != nil {
+					if errors.Is(err, sqlbase.ErrDescriptorNotFound) {
+						return false, nil
+					}
+					return false, err
+				}
+				typ, err = createTypeFromDesc(ctx, p, typDesc)
+				if err != nil {
+					return false, err
+				}
+				if err := addPGTypeRow(h, nspOid, typ, addRow); err != nil {
+					return false, err
+				}
+
+				// No errors and matches.
+				return false, nil
+			},
+		},
+	},
+}
+
+func createTypeFromDesc(ctx context.Context, p *planner, typDesc *sqlbase.ImmutableTypeDescriptor) (*types.T, error) {
```

Comment:
[nit] I don't think there is much benefit in pulling this out into a function. Coincidentally, the linter is failing here -- you need to gofmt this, the line is too long

Reply:
Got it. Reverted.